javascript/javascript2/week2/script1.js

```diff
@@ -0,0 +1,85 @@
+//Ex-1 Doubling of number
+//doubles the odd numbers in an array and throws away the even number.
+
+let numbers = [1, 2, 3, 4];
+let newNumbers = numbers
+    .filter(num => num % 2 !== 0)
+    .map(num => num * 2);
+
+console.log("The doubled numbers are", newNumbers); // [2, 6]
+
+//Ex-2 Working with movies
+//1. Create an array of movies containing the movies with a short title (you define what short means)
+const shortTitleMovies = movies.filter(shortMovie => shortMovie.title.length < 5);
+console.log(shortTitleMovies);
+
+//2.Create an array of movie titles with long movie titles
+const longTitleMovies = movies.filter(longMovie => longMovie.title.length > 25);
+console.log(longTitleMovies);
+
+//3. Count the number of movies made between 1980-1989 (including both the years)
+const oldMovies = movies
+    .filter(movie => movie.year >= 1980)
+    .filter(movie => movie.year <= 1989)
+console.log(`${oldMovies.length} Movies were made between 1980-1989 (including both the years)`);
+
+//4.Create a new array that has an extra key called tag. 
+//The tag is based on the rating: Good (>= 7), Average (>= 4 and < 7), Bad (< 4)
+
+const moviesWithTag = [];
+const goodMovie = movies.filter(movie => movie.rating >= 7);
+goodMovie.forEach(movie => movie.tag = "Good");
```

Comment:
I think you can do const goodMovie = movies.filter(movie => movie.rating >= 7).forEach(movie => movie.tag = "Good"); with a dot.